src/dft/backends/rocfft/commit.cpp

```diff
@@ -0,0 +1,427 @@
+            if constexpr (dom == dft::domain::COMPLEX) {
+                return rocfft_transform_type_complex_inverse;
+            }
+            else {
+                return rocfft_transform_type_real_inverse;
+            }
+        }();
+
+        constexpr rocfft_precision precision = [] {
+            if constexpr (prec == dft::precision::SINGLE) {
+                return rocfft_precision_single;
+            }
+            else {
+                return rocfft_precision_double;
+            }
+        }();
+
+        const std::size_t dimensions = config_values.dimensions.size();
+
+        constexpr std::size_t max_supported_dims = 3;
+        std::array<std::size_t, max_supported_dims> lengths;
+        // rocfft does dimensions in the reverse order to oneMKL
+        std::copy(config_values.dimensions.crbegin(), config_values.dimensions.crend(),
+                  lengths.data());
+
+        const std::size_t number_of_transforms =
+            static_cast<std::size_t>(config_values.number_of_transforms);
+
+        const std::size_t fwd_distance = static_cast<std::size_t>(config_values.fwd_dist);
+        const std::size_t bwd_distance = static_cast<std::size_t>(config_values.bwd_dist);
+
+        const rocfft_array_type fwd_array_ty = [&config_values]() {
+            if constexpr (dom == dft::domain::COMPLEX) {
+                if (config_values.complex_storage == dft::config_value::COMPLEX_COMPLEX) {
+                    return rocfft_array_type_complex_interleaved;
+                }
+                else {
+                    return rocfft_array_type_complex_planar;
+                }
+            }
+            else {
+                return rocfft_array_type_real;
+            }
+        }();
+        const rocfft_array_type bwd_array_ty = [&config_values]() {
+            if constexpr (dom == dft::domain::COMPLEX) {
+                if (config_values.complex_storage == dft::config_value::COMPLEX_COMPLEX) {
+                    return rocfft_array_type_complex_interleaved;
+                }
+                else {
+                    return rocfft_array_type_complex_planar;
+                }
+            }
+            else {
+                if (config_values.conj_even_storage != dft::config_value::COMPLEX_COMPLEX) {
+                    throw mkl::exception(
+                        "dft/backends/rocfft", __FUNCTION__,
+                        "only COMPLEX_COMPLEX conjugate_even_storage is supported");
+                }
+                return rocfft_array_type_hermitian_interleaved;
+            }
+        }();
+
+        std::array<std::size_t, 2> in_offsets{
+            static_cast<std::size_t>(config_values.input_strides[0]),
+            static_cast<std::size_t>(config_values.input_strides[0])
+        };
+        std::array<std::size_t, 2> out_offsets{
+            static_cast<std::size_t>(config_values.output_strides[0]),
+            static_cast<std::size_t>(config_values.output_strides[0])
+        };
```

Comment:
This is surprising to me. Shouldn't the offsets be set to 0 always?

Reply:
In all our test cases output_strides will be 0

Reply:
After talking offline this makes sense. stride[0] is the displacement in oneMKL whereas this is a separate field in rocfft.